usr/lib/soft_stdll/soft_specific.c

```diff
@@ -4365,6 +4365,19 @@ static CK_RV fill_ec_key_from_pubkey(EC_KEY *ec_key, const CK_BYTE *data,
         goto out;
     }
 
+    if (EC_POINT_is_on_curve(EC_KEY_get0_group(ec_key),
+                             EC_KEY_get0_public_key(ec_key), NULL) != 1) {
```

Comment:
This check is also done by `EC_KEY_check_type` as far as I can see in the code.  However, since this is not documented (or I just could not find it), I would keep this check here and below.

Reply:
Right, that was also my thinking when I coded that. `EC_KEY_check_key` should actually also check if the public key point is on the curve, but the documentation does not state that explicitly. So I also thought that its better to leave it in, even if the same thing is checked twice.